File: tests/perf/array_index_iter.py
#!/usr/bin/env python
""" Tests iteration over array using explicit index. """

def run():
    """ Body of the test """

    loops = 1000
    size = 10000
    elements = list(range(size))
    total = 0

    loop_idx = 0
    while loop_idx < loops:
        loop_idx += 1

        i = 0
        while i < size:
            elements[i] = i
            i += 1

        i = 0
        while i < size:
            total += elements[i]
            i += 1

    print(total)

run()

File: tests/perf/array_index_iter.js
const loops = 1000;
const size  = 10000;
const a     = [];
let   total = 0;

for (let l = 0; l < loops; l++) {
    for (let i = 0; i < size; i++) {
        a[i] = i;
    }

    for (let i = 0; i < size; i++) {
        total += a[i];
    }
}

console.log(total);
